### PR TITLE
fix: type inference fails when not using a transformer

### PR DESCRIPTION
**Pull request by juliusmarminge**
Regression from #3261 

Inferred types were serialized even when there wasn't a transformer defined

Same bug as found here: https://github.com/trpc/trpc/pull/3298#pullrequestreview-1200729879

### CODE REVIEW

File: packages/server/src/core/types.ts [hunk 1]
```diff
@@ -59,7 +60,7 @@ type GetInferenceHelpers<
       : TRouterOrProcedure extends AnyProcedure
       ? TType extends 'input'
         ? inferProcedureInput<TRouterOrProcedure>
-        : inferProcedureOutput<TRouterOrProcedure>
+        : inferTransformedProcedureOutput<TRouterOrProcedure>
```
Comment (KATT) [inline packages/server/src/core/types.ts:63]: This is a breaking change in another way than the other issue.

If you have an integration test or use the ssg helper, the data is unchanged.

I don't have a better idea right now but I am not confident this is the right solution

Comment (KATT) [reply packages/server/src/core/types.ts:63]: This was actually intentional b/c I didn't want a breaking change

Comment (sachinraja) [reply packages/server/src/core/types.ts:63]: I agree - we should not change this. If someone wants the transformed type they can use `inferTransformedProcedureOutput` directly (for now, at least).

**Review by KATT — APPROVED** [review]
YOLO

I think it's very very unlikely that anyone is using the output inference helper and get headaches here.

**Comment by KATT** [comment]
tl;dr - this **is** a minor breaking change but I rather see it as a bug fix since we're < 2 weeks post-release and I doubt it will cause issues for anyone
